spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/pubsub/GcpPubSubAutoConfiguration.java

```diff
@@ -90,6 +94,8 @@
 
 	private final HeaderProvider headerProvider = new UserAgentHeaderProvider(this.getClass());
 
+	private final Logger logger = LoggerFactory.getLogger(this.getClass());
```

Comment:
We usually have a static logger with a hardcoded class name. This prevents multiple instances of the same logger for each object instance (which is not really a concern for an autoconfiguration object, which is a singleton, but it's good to have consistency across the codebase).

Sonar will have an opinion on this, too -- https://rules.sonarsource.com/java/tag/convention/RSPEC-1312

Reply:
That makes sense, done.